pkg/cloud/services/compute/instance.go

```diff
@@ -182,17 +182,20 @@ func (s *Service) constructPorts(openStackCluster *infrav1.OpenStackCluster, ins
 
 	// no networks or ports found in the spec, so create a port on the cluster network
 	if len(ports) == 0 {
-		ports = []infrav1.PortOpts{{
+		port := infrav1.PortOpts{
 			Network: &infrav1.NetworkFilter{
 				ID: openStackCluster.Status.Network.ID,
 			},
-			FixedIPs: []infrav1.FixedIP{{
+			Trunk: &instanceSpec.Trunk,
+		}
+		for _, subnet := range openStackCluster.Status.Network.Subnets {
+			port.FixedIPs = append(port.FixedIPs, infrav1.FixedIP{
```

Comment:
Can help me to understand, why we need all subnets in `FixedIPs`.

Reply:
This is only if we have explicitly defined multiple subnets in the cluster network. In a dual stack cluster we will have separate ipv4 and ipv6 subnets for the same network, and the port needs an address from both. In a single stack cluster this just does exactly the same as it always did.